### ISSUE TITLE
Current release incompatible with Python 3.8.0

### ISSUE THREAD

**Issue by DrLuke** [Tinche/cattrs#77]
* cattrs version: 0.9.0
* Python version: 3.8.0
* Operating System: Linux

### Description

I was running AWS CDK, which uses JSII, which uses cattr and the following error came up:
```
  File "/home/drluke/work/di-client-api-message-dispatcher/venv/lib/python3.8/site-packages/cattr/_compat.py", line 86, in <module>
    from typing import _Union
ImportError: cannot import name '_Union' from 'typing' (/usr/lib/python3.8/typing.py)

```

I looked into the file and found the following:
```python3
is_py37 = version_info[:2] == (3, 7)
```

which is then used to decide whether to import `Union` or `_Union`. The latter doesn't exist in python3.8's typing module, causing the import error.

My suggestion would be to check if we're on python3.7 or greater instead.
```python3
is_py37_or_greater = version_info[0] == 3 and version_info[1] >= 7
```

**Comment by nedbat** [comment]
```
is_py37_plus = version_info >= (3, 7)
```

**Comment by llchan** [comment]
FYI, some related links:
- Issue about the _Union import in general: #46
- PR that fixes 3.8: #73

**Comment by tirkarthi** [comment]
I guess this can be closed now since python 3.8 is supported with PR https://github.com/Tinche/cattrs/pull/73 but other problems are present in Python 3.9 similar to version check that I will create a new issue.
